python/ray/experimental/workflow/common.py

```diff
@@ -22,6 +23,19 @@
     from ray.experimental.workflow.storage import Storage
 
 
+@unique
+class WorkflowStatus(str, Enum):
+    # There is at least a remote task running in ray cluster
+    RUNNING = "RUNNING"
+    # It got canceled and can't be resumed later
+    CANCELED = "CANCELED"
+    # The step is finished. For virtual actor, it means that all
+    # writing steps has been finished
+    FINISHED = "FINISHED"
```

Comment:
Clarify this includes workflows that failed with application-level error?

Reply:
Do you mean `catch_exceptions=True` ?

Reply:
I guess even with catch exceptions false, if it finished with error this would be it's state, correct? (We don't distinguish between successfully finished and finished with error).